euphony/src/main/java/co/euphony/rx/EuRxManager.java

```diff
@@ -102,6 +106,15 @@ public boolean listen() {
 			return listenOnNative();
 	}
 
+	public boolean listen(Context context) { // Safely call listen function without audio permission.
+		int hasPermission = context.checkPermission(Constants.REQUIRED_PERMISSION, Process.myPid(), Process.myUid());
+		if (hasPermission == PERMISSION_DENIED) {
+			Log.e(LOG, "Before calling listen(), you should acquire RECORD_AUDIO permission.");
+			return false;
+		}
+		return listen();
+	}
```

Comment:
For checking permission, here is the important part in this PR :)
You did very well. I think it is very beautiful solution to resolve #205 

But, If there is one element that needs to be improved, 
it is the part that receives the Android `context` as an argument to the listen function.
![image](https://user-images.githubusercontent.com/1748714/188186971-4296d347-c507-4c94-abbf-34ab12cb7f40.png)
It means that `EuRxManager` class depends on `Context`. 

When designing a class, coupling can inevitably occur.
and even in this case, interfaces are often used for loose coupling. 

If the coupling can be removed, it is best to do so.

Therefore, what i'm suggested is to use `try` `catch` statements to resolve the #205.
Actually, The root problem is the start and stop functions of audioRecord that cause IllegalStateException.
https://github.com/euphony-io/euphony/blob/ca9985bd2dafbd1da10d552abf77c635eb449ab7/euphony/src/main/java/co/euphony/rx/AudioRecorder.java#L59
```java
try {
    audioRecord.startRecording();
    running = true;
} catch (IllegalStateException e) {
    running = false;
    Log.e(LOG, "Before calling listen(), you should acquire RECORD_AUDIO permission.");
}
```

https://github.com/euphony-io/euphony/blob/ca9985bd2dafbd1da10d552abf77c635eb449ab7/euphony/src/main/java/co/euphony/rx/AudioRecorder.java#L69
```java
try {
    audioRecord.stop();
    running = true;
} catch (IllegalStateException e) {
    running = false;
    Log.e(LOG, "Before calling listen(), you should acquire RECORD_AUDIO permission.");
}
```

Maybe it'll be same conclusion without `Context`
If you have any concerns, Let me know :)

Anyway, this PR will make euphony a more developer-friendly library!

refer to android soruce : https://android.googlesource.com/platform/frameworks/base/+/master/core/jni/android_media_AudioRecord.cpp#331